core/src/authorizations.ts

```diff
@@ -1,37 +1,56 @@
 import type { Notifier } from './types';
 import getContract, { getContractAddressByName, getClipperNameByCollateralType } from './contracts';
 import executeTransaction from './execute';
+import memoizee from 'memoizee';
+import getSigner from './signer';
 
-export const authorizeWallet = async function (
-    network: string,
-    revoke: boolean,
-    notifier?: Notifier
-): Promise<string> {
+const _authorizeWallet = async function (network: string, revoke: boolean, notifier?: Notifier): Promise<string> {
+    const signer = await getSigner(network);
     const joinDaiAddress = await getContractAddressByName(network, 'MCD_JOIN_DAI');
     const contractMethod = revoke ? 'nope' : 'hope';
-    return executeTransaction(network, 'MCD_VAT', contractMethod, [joinDaiAddress], notifier);
+    const transaction = await executeTransaction(network, 'MCD_VAT', contractMethod, [joinDaiAddress], notifier);
+    await getWalletAuthorizationStatus.delete(network, await signer.getAddress());
+    return transaction;
 };
 
-export const authorizeCollateral = async function (
+export const authorizeWallet = memoizee(_authorizeWallet, {
+    promise: true,
+    length: 3,
```

Comment:
You don't have to cache `notifier`, since no matter the notifier, the result would stay the same
```suggestion
    length: 2,
```